examples/extra_element_integer/random_cluster.cpp

```diff
@@ -0,0 +1,148 @@
+#include "libmesh/equation_systems.h"
+#include "libmesh/exodusII_io.h"
+#include "libmesh/libmesh.h"
+#include "libmesh/linear_implicit_system.h"
+#include "libmesh/mesh.h"
+#include "libmesh/mesh_generation.h"
+#include "libmesh/numeric_vector.h"
+#include "libmesh/system.h"
+
+#include <cstdlib>
+#include <ctime>
+#include <random>
+#include <vector>
+
+using namespace libMesh;
+
+void CreateMesh(libMesh::Mesh &mesh, int nx, int ny) {
+  libMesh::MeshTools::Generation::build_square(mesh, nx, ny, 0.0, 10.0, 0.0,
+                                               10.0);
+}
+
+void CreateMesh(libMesh::Mesh &mesh, int nx, int ny, int nz) {
+  libMesh::MeshTools::Generation::build_cube(mesh, nx, ny, nz, 0.0, 10.0, 0.0,
+                                             10.0, 0.0, 10.0);
+}
+
+void PopulateRandomIntegers(libMesh::Mesh &mesh, LinearImplicitSystem &system) {
+  DofMap &dof_map = system.get_dof_map();
+  for (const auto &elem : mesh.element_ptr_range()) {
+
+    std::vector<dof_id_type> dof_indices;
+    dof_map.dof_indices(elem, dof_indices);
+    int random_int = rand() % 5;
+    system.solution->set(dof_indices[0], 5 - random_int);
+  }
+}
+
+void AddVaribalesToSystem(libMesh::EquationSystems &equation_systems,
+                          LinearImplicitSystem &system,
+                          std::string var_name, bool needs_re_init =false) {
+    system.add_variable(var_name, CONSTANT, MONOMIAL);
+    if (!needs_re_init){
+        equation_systems.init();
+    }
+    else{
+        equation_systems.reinit();
+    }
+}
+
+void FindCluster(libMesh::Mesh &mesh, libMesh::DofMap &dof_map,
+                 libMesh::DofMap &local_dof_map, LinearImplicitSystem &system,
+                 LinearImplicitSystem &local_system,
+                 const std::string &variable_name,
+                 libMesh::EquationSystems &equation_systems,
+                 const unsigned int index) {
+
+  const unsigned int variable_num = system.variable_number(variable_name);
+
+  for (const auto &elem : mesh.element_ptr_range()) {
+
+    std::vector<dof_id_type> global_dof_indices;
+    dof_map.dof_indices(elem, global_dof_indices, variable_num);
+
+    std::vector<double> solution_value(1);
+    system.solution->get(global_dof_indices, solution_value);
+    int element_solution = static_cast<int>(solution_value[0]);
+
+    bool belong_to_a_cluster = false;
+    for (unsigned int side = 0; side < elem->n_sides(); ++side) {
+      const Elem *neighbor = elem->neighbor_ptr(side);
+      if (neighbor) {
+        std::vector<dof_id_type> neighbor_dof_indices;
+        dof_map.dof_indices(neighbor, neighbor_dof_indices, variable_num);
+
+        std::vector<double> neighbor_solution_value(1);
+        system.solution->get(neighbor_dof_indices, neighbor_solution_value);
+        int neighbor_element_solution =
+            static_cast<int>(neighbor_solution_value[0]);
+
+        if (element_solution == neighbor_element_solution) {
+          belong_to_a_cluster = true;
```

Comment:
why not set the `extra_integer` here?

Reply:
Need to find a way to number the clusters - this is TBD - but I don't think you should just use the solution as the cluster ID - it confuses what we ultimately need to do in the future.